javav2/example_code/ec2/src/main/java/com/example/ec2/FindRunningInstances.java

```diff
@@ -1,100 +1,91 @@
-                                    "AMI %s, " +
-                                    "type %s, " +
-                                    "state %s " +
-                                    "and monitoring state %s",
-                            instance.instanceId(),
-                            instance.imageId(),
-                            instance.instanceType(),
-                            instance.state().name(),
-                            instance.monitoring().state());
-                    System.out.println("");
-                }
-            }
-            nextToken = response.nextToken();
-
-            } while (nextToken != null);
-
-        } catch (Ec2Exception e) {
-           System.err.println(e.awsErrorDetails().errorMessage());
-           System.exit(1);
-        }
-        // snippet-end:[ec2.java2.running_instances.main]
-    }
-}
+// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
+//snippet-sourcedescription:[FindRunningInstances.java demonstrates how to find running Amazon Elastic Compute Cloud (Amazon EC2) instances by using a filter.]
+//snippet-keyword:[AWS SDK for Java v2]
+//snippet-keyword:[Code Sample]
+//snippet-service:[Amazon EC2]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[11/01/2020]
+//snippet-sourceauthor:[scmacdon-aws]
+
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.ec2;
+
+// snippet-start:[ec2.java2.running_instances.import]
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.Filter;
+import software.amazon.awssdk.services.ec2.model.DescribeInstancesRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeInstancesResponse;
+import software.amazon.awssdk.services.ec2.model.Reservation;
+import software.amazon.awssdk.services.ec2.model.Instance;
+import software.amazon.awssdk.services.ec2.model.Ec2Exception;
+// snippet-end:[ec2.java2.running_instances.import]
+
+/**
+ * Locates all running EC2 instances using a Filter.
+ */
+public class FindRunningInstances {
+    public static void main(String[] args) {
+
+        Region region = Region.US_WEST_2;
+        Ec2Client ec2 = Ec2Client.builder()
+                .region(region)
+                .build();
+
+        findRunningEC2Instances(ec2);
+        ec2.close();
+    }
+
+   // snippet-start:[ec2.java2.running_instances.main]
+   public static void findRunningEC2Instances(Ec2Client ec2) {
+
+       try {
+
+           String nextToken = null;
+
+           do {
+                // Create a Filter to find all running instances
+                Filter filter = Filter.builder()
+                    .name("instance-state-name")
+                    .values("running")
+                    .build();
+
+                //Create a DescribeInstancesRequest

```

Comment:
Delete this comment as it is not adding any information that isn't readily apparent.

Reply:
Fixed

Reply:
Fixed